File: algorithm/55. Jump Game/55.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int step=0,max=nums.size();
        for(int i=0; i< nums.size()-1;i++)
        {
            step--;
            if(nums[i]>step)
                step=nums[i];
            if(step==0)
                return false;
        }
        return true;
    }
};

File: algorithm/55. Jump Game/55.java
class Solution {
    public boolean canJump(int[] nums) {
        int count = 0;
        for(int i = 0; i <nums.length; i++)
        {
            if(nums[i] >= count)
                count = nums[i];
            else
            {
                count--;
            }
            if(count == 0 && i <nums.length - 1)
                return false;
        }
        return true;
    }
}
